apps/gallery/js/ImageEditor.js

```diff
@@ -298,10 +300,16 @@ function editPhoto(n) {
     var imagesize = metadata.width * metadata.height;
     var maxsize = CONFIG_MAX_EDIT_PIXEL_SIZE || CONFIG_MAX_IMAGE_PIXEL_SIZE;
 
+    LazyLoader.load(['shared/js/media/jpeg-exif.js'], () => {
+      JPEGParser.readExifMetaData(file, (error, metaData) => {
+        editSettings.EXIF = metaData;
+      });
+    });
+
     if (metadata.rotation || metadata.mirrored || imagesize > maxsize) {
       Spinner.show();
       cropResizeRotate(file, null, maxsize || null,
-                       null, metadata,
+                       'image/jpeg+exif', metadata,
```

Comment:
I think you only want to pass image/jpeg+exif if the file is already a jpeg. Otherwise use null to leave png files unchanged.  

Note that this code can be triggered for large PNGs on low memory devices. This is not only an EXIF rotation thing.